src/context.js

```diff
@@ -104,13 +122,22 @@ export class Context {
 			throw new Error(message);
 		}
 
+		if (node.start && this.#sourceText) {
```

Comment:
Could fail if start is 0 (same below for `node.end` though that's even less likely)